// react-native-simple-banner-carousel/lib/components/Card.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Card = void 0;
const react_1 = __importDefault(require("react"));
const react_native_1 = require("react-native");
function Card({ id, source, width, title, description, score, style }) {
    return (react_1.default.createElement(react_native_1.View, { key: id, style: [
            {
                width: width,
                maxWidth: '100%',
                backgroundColor: '#fffffff',
                paddingTop: 12,
                paddingBottom: 16,
            },
            style,
        ] },
        react_1.default.createElement(react_native_1.Image, { source: source, style: {
                width: width,
                maxWidth: '100%',
                height: 156,
                backgroundColor: '#bbb',
                marginBottom: 8,
            } }),
        react_1.default.createElement(react_native_1.View, { style: {
                display: 'flex',
                flexDirection: 'row',
                justifyContent: 'space-between',
            } },
            react_1.default.createElement(react_native_1.Text, { style: {
                    flex: 11,
                    fontSize: 14,
                } }, title.length > 25 ? `${title.slice(0, 25)}...` : title),
            react_1.default.createElement(react_native_1.View, { style: {
                    flex: 1,
                    display: 'flex',
                    justifyContent: 'center',
                    alignItems: 'center',
                    borderRadius: 25,
                    paddingVertical: 6,
                    paddingHorizontal: 4,
                    backgroundColor: '#f0f0f0',
                } },
                react_1.default.createElement(react_native_1.Text, { style: {
                        fontSize: 12,
                    } }, score))),
        react_1.default.createElement(react_native_1.View, { style: {
                display: 'flex',
                flexDirection: 'row',
                justifyContent: 'space-between',
            } },
            react_1.default.createElement(react_native_1.Text, { style: {
                    fontSize: 12,
                } }, description))));
}
exports.Card = Card;
