// react-native-simple-banner-carousel/lib/components/Indicator.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Indicator = void 0;
const react_1 = __importDefault(require("react"));
const react_native_1 = require("react-native");
const Dot_1 = require("./Dot");
function Indicator({ data, currentIndex, lastIndex }) {
    return (react_1.default.createElement(react_native_1.View, { style: {
            display: 'flex',
            flexDirection: 'row',
            justifyContent: 'center',
            alignItems: 'center',
            position: 'absolute',
            bottom: '5%',
            left: 0,
            width: react_native_1.Dimensions.get('window').width,
            height: 24,
            backgroundColor: 'transparent',
            zIndex: 1,
        } }, data.map((_, i) => (react_1.default.createElement(Dot_1.Dot, { key: `simple_carousel_indicator_key_${i}`, index: i, currentIndex: currentIndex, lastIndex: lastIndex })))));
}
exports.Indicator = Indicator;
